### PR TITLE
make TestHTTPServer choose the port automatically rather than guessing

### PR DESCRIPTION
**Pull request by weissi**
the loop to find a free port is ugly, error-prone and makes the tests less robust

    let start = 21961
    for port in start...(start+100) { //we must find at least one port to bind

fixing this here

### CODE REVIEW

File: TestFoundation/HTTPServer.swift [hunk 1]
```diff
@@ -35,6 +35,27 @@ struct _HTTPUtils {
...
+            ptr.withMemoryRebound(to: UInt8.self, capacity: MemoryLayout<UInt16>.size) { data in
+                var result: UInt16 = 0
+                result |= (UInt16(data[1]) << 0)
+                result |= (UInt16(data[0]) << 8)
+                return result
+            }
+        }
+    }
+}
+
```
Comment (spevans) [inline TestFoundation/HTTPServer.swift]: isn't this just `UInt16(bigEndian:)` ?

File: TestFoundation/HTTPServer.swift [hunk 2]
```diff
@@ -35,6 +35,27 @@ struct _HTTPUtils {
...
+private extension UInt16 {
+    /* this takes an UInt16 in network byte order (big-endian) representation and converts it to
+     whatever endianness the we're running on. It does work for both big
+     and little endian machines.
+     Cf. https://commandcenter.blogspot.co.uk/2012/04/byte-order-fallacy.html
+
+     this is the same as nthos which is a C macro which isn't available from Swift :\.
+     */
+    static func from(networkByteOrder input: UInt16) -> UInt16 {
+        var val: UInt16 = input
```
Comment (spevans) [inline TestFoundation/HTTPServer.swift]: Isn't this just `UInt16(bigEndian:)` ?

File: TestFoundation/HTTPServer.swift [hunk 3]
```diff
@@ -35,6 +35,27 @@ struct _HTTPUtils {
     static let EMPTY = ""
 }
 
+private extension UInt16 {
+    /* this takes an UInt16 in network byte order (big-endian) representation and converts it to
+     whatever endianness the we're running on. It does work for both big
+     and little endian machines.
+     Cf. https://commandcenter.blogspot.co.uk/2012/04/byte-order-fallacy.html
+
+     this is the same as nthos which is a C macro which isn't available from Swift :\.
```
Comment (pushkarnk) [inline TestFoundation/HTTPServer.swift]: a small typo here? did you mean `ntohs`?

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
```
Test Case 'TestURLSession.test_httpRedirectionTimeout' started at 2017-08-23 07:52:21.900
TestFoundation/TestURLSession.swift:325: error: TestURLSession.test_httpRedirectionTimeout : XCTAssertEqual failed: ("cannotConnectToHost") is not equal to ("timedOut") - Unexpected error code
```

File: TestFoundation/HTTPServer.swift (hunk 2 above)
Comment (weissi) [inline TestFoundation/HTTPServer.swift]: yes, thanks, fixed.

File: TestFoundation/HTTPServer.swift (hunk 3 above)
Comment (weissi) [inline TestFoundation/HTTPServer.swift]: thanks, removed as of @spevans comment

**Comment by weissi** [comment]
@swift-ci please test

**Comment by weissi** [comment]
@swift-ci please test

**Comment by weissi** [comment]
@pushkarnk @spevans I'm actually expect the tests to fail in `test_httpRedirectionTimeout` just I'd also expect them to fail in #1187 . If you're happy I'd temporarily disable `test_httpRedirectionTimeout` and file an issue to fix it. I volunteer to do that.

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Comment by pushkarnk** [comment]
cc @saiHemak

**Pull request merged**